torch/distributed/fsdp/_state_dict_utils.py

```diff
@@ -706,55 +706,40 @@ def _post_load_state_dict_hook(
     _post_load_state_dict_hook_fn[fsdp_state._state_dict_type](module, fsdp_state)
 
 
-@no_type_check
-def _register_state_dict_pre_hooks(state: _FSDPState) -> None:
-    """Registers the pre-save state dict hooks."""
-    if not _is_composable(state):
-        state.register_state_dict_pre_hook(
-            functools.partial(_pre_state_dict_hook, state)
-        )
-        return
-    for handle in state._handles:
-        handle._comm_module.register_state_dict_pre_hook(
-            functools.partial(_pre_state_dict_hook, state)
-        )
-
-
-@no_type_check
-def _register_state_dict_hooks(state: _FSDPState) -> None:
-    """Registers the post-save state dict hooks."""
-    if not _is_composable(state):
-        state._register_state_dict_hook(functools.partial(_post_state_dict_hook, state))
-        return
-    for handle in state._handles:
-        handle._comm_module._register_state_dict_hook(
-            functools.partial(_post_state_dict_hook, state)
-        )
-
-
-@no_type_check
-def _register_load_state_dict_pre_hooks(state: _FSDPState) -> None:
-    """Registers the pre-load state dict hooks."""
-    if not _is_composable(state):
-        state._register_load_state_dict_pre_hook(
-            functools.partial(_pre_load_state_dict_hook, state), with_module=True
-        )
-        return
-    for handle in state._handles:
-        handle._comm_module._register_load_state_dict_pre_hook(
-            functools.partial(_pre_load_state_dict_hook, state), with_module=True
+def _register_all_state_dict_hooks(state: _FSDPState):
+    """
+    Registers pre-save, post-save, pre-load, and post-load state dict hooks.
+    """
+    for hook_registration_fn_str, hook, hook_registration_fn_kwargs in (
+        ("register_state_dict_pre_hook", _pre_state_dict_hook, {}),
+        ("_register_state_dict_hook", _post_state_dict_hook, {}),
+        (
+            "_register_load_state_dict_pre_hook",
+            _pre_load_state_dict_hook,
+            {"with_module": True},
+        ),
+        ("register_load_state_dict_post_hook", _post_load_state_dict_hook, {}),
+    ):
+        _register_state_dict_hooks_base(
+            state, hook_registration_fn_str, hook, hook_registration_fn_kwargs
         )
 
 
 @no_type_check
-def _register_load_state_dict_post_hooks(state: _FSDPState) -> None:
-    """Registers the post-load state dict hooks."""
+def _register_state_dict_hooks_base(
+    state: _FSDPState,
+    hook_registration_fn_name: str,
+    hook: Callable,
+    hook_registration_fn_kwargs: Dict[str, Any],
+) -> None:
+    """Registers ``hook`` using ``hook_registration_fn``."""
+    hook_with_state = functools.partial(hook, state)
```

Comment:
nit: maybe add a TODO and follow up issue to get state directly from module and avoid this partial here. cc @fegin

Reply:
Done.